### ISSUE TITLE
Unexpected behavior with cnoremap and wildcharm

### ISSUE THREAD

**Issue by gpanders** [vim/vim#4954]
**Describe the bug**
Certain cnoremap mappings seem to not work when `'wildcharm'` is set to `<C-Z>` but do work when set to `<Tab>`.

**To Reproduce**
Minimal vimrc:
```vim
set nocompatible
set noloadplugins

set wildmenu
set wildcharm=<C-Z>
cnoremap <expr> <C-J> wildmenumode() ? "\<Down>\<C-Z>" : "\<C-J>"
```

**Expected behavior**
Pressing `<C-J>`  while on a directory in the wild menu should "open" that directory and restart wild mode completion.

**Actual behavior**
Vim inserts a `^I` character into the command line and does nothing further.

However, if I instead use
```vim
set wildcharm=<Tab>
cnoremap <expr> <C-J> wildmenumode() ? "\<Down>\<Tab>" : "\<C-J>"
```

It works as expected. My understanding is that `<C-Z>` should act just like `<Tab>` in a mapping when `cnoremap=<C-Z>`, but that does not seem to be the case.

**Environment (please complete the following information):**
```
$ vim --version
VIM - Vi IMproved 8.1 (2018 May 18, compiled Jun 11 2019 08:22:43)
macOS version
Included patches: 1-1517
Compiled by travis@Traviss-Mac.local
Huge version with MacVim GUI.  Features included (+) or not (-):
+acl               +file_in_path      +mouse_urxvt       -tcl
+arabic            +find_in_path      +mouse_xterm       +termguicolors
+autocmd           +float             +multi_byte        +terminal
+autochdir         +folding           +multi_lang        +terminfo
-autoservername    -footer            -mzscheme          +termresponse
+balloon_eval      +fork()            +netbeans_intg     +textobjects
+balloon_eval_term +fullscreen        +num64             +textprop
+browse            -gettext           +odbeditor         +timers
++builtin_terms    -hangul_input      +packages          +title
+byte_offset       +iconv             +path_extra        +toolbar
+channel           +insert_expand     +perl/dyn          +transparency
+cindent           +job               +persistent_undo   +user_commands
+clientserver      +jumplist          +postscript        +vartabs
+clipboard         +keymap            +printer           +vertsplit
+cmdline_compl     +lambda            +profile           +virtualedit
+cmdline_hist      +langmap           +python/dyn        +visual
+cmdline_info      +libcall           +python3/dyn       +visualextra
+comments          +linebreak         +quickfix          +viminfo
+conceal           +lispindent        +reltime           +vreplace
+cryptv            +listcmds          +rightleft         +wildignore
+cscope            +localmap          +ruby/dyn          +wildmenu
+cursorbind        +lua/dyn           +scrollbind        +windows
+cursorshape       +menu              +signs             +writebackup
+dialog_con_gui    +mksession         +smartindent       -X11
+diff              +modify_fname      -sound             -xfontset
+digraphs          +mouse             +spell             +xim
+dnd               +mouseshape        +startuptime       -xpm
-ebcdic            +mouse_dec         +statusline        -xsmp
+emacs_tags        -mouse_gpm         -sun_workshop      -xterm_clipboard
+eval              -mouse_jsbterm     +syntax            -xterm_save
+ex_extra          +mouse_netterm     +tag_binary
+extra_search      +mouse_sgr         -tag_old_static
-farsi             -mouse_sysmouse    -tag_any_white
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/Applications/MacVim.app/Contents/Resources/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_MACVIM -Wall -Wno-unknown-pragmas -pipe  -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -fstack-protector-strong -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib -L. -fstack-protector-strong -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o Vim -framework Cocoa -framework Carbon       -lm  -lncurses -liconv -framework AppKit   -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE
```

OS: macOS Mojave

**Comment by bfrg** [comment]
It works when you replace `<Down>` with `<C-E>`:
```vim
set wildcharm=<C-Z>
cnoremap <expr> <C-J> wildmenumode() ? "\<C-E>\<C-Z>" : "\<C-J>"
```

**Comment by gpanders** [comment]
Another data point: the converse mapping (`<C-K>`) works just as expected:

```vim
cnoremap <expr> <C-K> wildmenumode() ? "\<Up>\<C-Z>" : "\<C-K>"
```

**Comment by lacygoill** [comment]
Fixed by [8.2.2221](https://github.com/vim/vim/releases/tag/v8.2.2221).
